sled-agent/src/updates.rs

```diff
@@ -220,6 +223,7 @@ impl UpdateManager {
             let entry = entry.map_err(|err| io_err(dir, err))?;
             let file_type =
                 entry.file_type().map_err(|err| io_err(dir, err))?;
+            let path: Utf8PathBuf = entry.path().try_into()?;
```

Comment:
Definitely worth using `read_dir_utf8` here.

Reply:
Done in https://github.com/oxidecomputer/omicron/pull/2988/commits/ced5dc92aa41c74da6a846aaa9cd59d20d9c7843